build/helper/codegen_helper.py

```diff
@@ -295,31 +298,73 @@ def _get_ctype_variable_definition_snippet_for_buffers(parameter, parameters, iv
         elif parameter['size']['mechanism'] == 'ivi-dance':
             if ivi_dance_step == IviDanceStep.QUERY_SIZE:
                 definition = 'None  # case B580'
             elif ivi_dance_step == IviDanceStep.GET_DATA:
                 size_parameter = find_size_parameter(parameter, parameters)
-                definition = '({0}.{1} * {2}.value)()  # case B590'.format(module_name, parameter['ctypes_type'], size_parameter['ctypes_variable_name'])
+                if parameter['is_array']:
+                    size_declaration = '{0}_size = {1}.value  # case B590'.format(parameter['python_name'], size_parameter['ctypes_variable_name'])
+                    array_declaration = '{0}_array = array.array("{1}", [0] * {0}_size)  # case B590'.format(parameter['python_name'], get_array_type_for_api_type(parameter['ctypes_type']))
+                    ctypes_declaration = '{0} = _converters.convert_iterable_to_ctypes({2}_array, ({1}.{3}))  # case B590'.format(parameter['ctypes_variable_name'], module_name, parameter['python_name'], parameter['ctypes_type'])
+                    definitions.append(size_declaration)
+                    definitions.append(array_declaration)
+                    definitions.append(ctypes_declaration)
+                elif parameter['is_list']:
+                    definition = '({0}.{1} * {2}.value)()  # case B590'.format(module_name, parameter['ctypes_type'], size_parameter['ctypes_variable_name'])
+                else:
+                    assert False, "Expected either 'is_array' or 'is_list' to be True. Both False."
             else:
                 assert False, "ivi_dance_step {0} not valid for parameter {1} with ['size']['mechanism'] == 'ivi-dance'".format(ivi_dance_step, parameter['name'])
         elif parameter['size']['mechanism'] == 'passed-in':
             size_parameter = find_size_parameter(parameter, parameters)
-            definition = '({0}.{1} * {2})()  # case B600'.format(module_name, parameter['ctypes_type'], size_parameter['python_name'])
+            if parameter['is_array']:
+                size_declaration = '{0}_size = {1}  # case B600'.format(parameter['python_name'], size_parameter['python_name'])
+                array_declaration = '{0}_array = array.array("{1}", [0] * {0}_size)  # case B600'.format(parameter['python_name'], get_array_type_for_api_type(parameter['ctypes_type']))
+                ctypes_declaration = '{0} = _converters.convert_iterable_to_ctypes({2}_array, ({1}.{3}))  # case B600'.format(parameter['ctypes_variable_name'], module_name, parameter['python_name'], parameter['ctypes_type'])
+                definitions.append(size_declaration)
+                definitions.append(array_declaration)
+                definitions.append(ctypes_declaration)
+            elif parameter['is_list']:
+                definition = '({0}.{1} * {2})()  # case B600'.format(module_name, parameter['ctypes_type'], size_parameter['python_name'])
```

Comment:
This function is pretty weirdly written.

Specifically, you exist the nested if/else structure of death and sometimes `definition` is defined in which case we still need to append something to the left of it and stick it inside `definitions`... or sometimes `definitions` is ready to go.

I think it should be done consistently, which is to say: whenever you exit the nested if/else structure of death definitions is already populated and ready to go.

Reply:
Every case exits with `definition` set. Some cases will have added more lines.